kai/reactive_codeplanner/task_runner/analyzer_lsp/validator.py

```diff
@@ -111,6 +111,7 @@ def __parse_analyzer_lsp_output(
                 violation = r.rulesets[key].violations[violationKey]
                 violation.incidents.sort()
                 for i in violation.incidents:
+                    violation.id = violationKey
```

Comment:
Architectural question, but is there any reason we're not using the [Report class](https://github.com/konveyor/kai/blob/4b6f35fcffd9797278807effe15b8dfeab81d6db/kai/analyzer_types.py#L204) for this function?

Reply:
think we already are. But we are sorting stuff after the fact here